zkevm-circuits/src/evm_circuit/witness.rs

```diff
@@ -65,10 +65,7 @@ impl BlockContext {
                 [
                     F::from(BlockContextFieldTag::Number as u64),
                     F::zero(),
-                    RandomLinearCombination::random_linear_combine(
-                        self.number.to_le_bytes(),
-                        randomness,
-                    ),
+                    self.number.to_scalar().unwrap(),
```

Comment:
In the spec, the Block table, in the Number tag contains the RLC enoded value of the block number: https://github.com/appliedzkp/zkevm-specs/blob/657d0b034932ea8fa6515c3ec1f9d1e0055f43b0/src/zkevm_specs/evm/typing.py#L63

I think it's better to assign the value without RLC encoding, like you do in this PR (using the assumption that the number will always fit in the field).  If I remember correctly, Han mentioned that the yellow paper doesn't specify a bound on the block number, but I think 64 bits is a sensible bound, and there have been attempts to make these bounds official: https://eips.ethereum.org/EIPS/eip-1985 (we discussed that we may adhere to this EIP in the zkEVM implementation).

To make the implementation match the spec, I think the spec should be updated so that the block number in the Block table is not RLC encoded.

Reply:
Thank you very much for explaining the detailed RLC implementation design background.

Reply:
Thank you very much for explaining the detailed RLC implementation design background.